### PR TITLE
OH2-202: Create backend for users settings

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-202.

### CODE REVIEW

File: src/main/java/org/isf/menu/rest/UserSettingController.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		} else {
+			LOGGER.info("create a UserSetting");
+			userSetting = userSettingMapper.map2Model(userSettingDTO);
+			userSetting.setUser(userName);
+			isCreated = userSettingManager.newUserSetting(userSetting);
+		}
+		
+		if (isCreated == null) {
+			LOGGER.info("UserSetting is not created!");
+            throw new OHAPIException(new OHExceptionMessage("UserSetting not created."));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/rest/UserSettingController.java:88]: the formatting from here to the end of the method looks wrong.

File: src/main/java/org/isf/menu/rest/UserSettingController.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		LOGGER.info("Attempting to fetch the UserSetting of dashboard of the current user");
+		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		String dashboard = "dashboard";
+		UserSetting userSetting = userSettingManager.getUserSettingDashboard(userName, dashboard);
+		if (userSetting == null) {
+			LOGGER.info("No settings for the current user");
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		} else {
+			UserSettingDTO userSettingDashboard = userSettingMapper.map2DTO(userSetting);
+	        LOGGER.info("Found {} user settings", userSettingDashboard);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/rest/UserSettingController.java:110]: Formatting needs fixing.

File: src/main/java/org/isf/menu/rest/UserSettingController.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	 * @return {@link UserSettingDTO} if the userSetting of configName dashboard exist, null otherwise.
+	 */
+	@GetMapping(value = "/usersettings/dashboard", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> getUserSettingDashboard() throws OHServiceException {
+		LOGGER.info("Attempting to fetch the UserSetting of dashboard of the current user");
+		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		String dashboard = "dashboard";
+		UserSetting userSetting = userSettingManager.getUserSettingDashboard(userName, dashboard);
+		if (userSetting == null) {
+			LOGGER.info("No settings for the current user");
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/rest/UserSettingController.java:106]:
```suggestion
			LOGGER.info("No dashboard settings for the current user {}", userName);
```

File: src/main/java/org/isf/menu/rest/UserSettingController.java (hunk 1 above)
Comment (ArnaudFonzam) [reply src/main/java/org/isf/menu/rest/UserSettingController.java:88]: please i don't know why we have this formatting here in my IDE on don't have it and I don't how I can solve it .

Comment (mwithi) [reply src/main/java/org/isf/menu/rest/UserSettingController.java]: If you follow https://github.com/informatici/openhospital-api#code-style you should be ready to go

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

File: src/main/java/org/isf/menu/rest/UserSettingController.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	 * @return {@link UserSettingDTO} if the userSetting has been inserted, null
+	 *         otherwise.
+	 * @throws OHServiceException
+	 */
+	@PostMapping(value = "/usersettings", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> newUserSetting(@Valid @RequestBody UserSettingDTO userSettingDTO)
+			throws OHServiceException {
+		LOGGER.info("Attempting to create or update a UserSetting");
+		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		UserSetting userSetting = userSettingManager.getUserSettingDashboard(userName, userSettingDTO.getConfigName());
```
Comment (mwithi) [inline src/main/java/org/isf/menu/rest/UserSettingController.java:76]:
```suggestion
		UserSetting userSetting = userSettingManager.getUserSetting(userName, userSettingDTO.getConfigName());
```

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**